common/src/main/java/io/homeassistant/companion/android/common/sensors/SensorManager.kt

```diff
@@ -42,7 +43,15 @@ interface SensorManager {
 
     fun checkPermission(context: Context, sensorId: String): Boolean {
         return requiredPermissions(sensorId).all {
-            context.checkPermission(it, myPid(), myUid()) == PackageManager.PERMISSION_GRANTED
+            if (sensorId != "last_used_app")
+                context.checkPermission(it, myPid(), myUid()) == PackageManager.PERMISSION_GRANTED
+            else {
+                val pm = context.packageManager
+                val appInfo = pm.getApplicationInfo(context.packageName, 0)
+                val appOpsManager = context.getSystemService(Context.APP_OPS_SERVICE) as AppOpsManager
+                val mode = appOpsManager.checkOpNoThrow(AppOpsManager.OPSTR_GET_USAGE_STATS, appInfo.uid, appInfo.packageName)
+                mode == AppOpsManager.MODE_ALLOWED
+            }
```

Comment:
When happens when we try to check for the permission like we check for the other?

Reply:
it continues to return false when we check for the permission in the manifest, its strange honestly but I think because its a protected permission? I was fighting with our permission check until I added t his in :)